src/main/resources/META-INF/reaktivity/socks.idl

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2016-2017 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+scope socks
+{
+    union SocksAddress switch (uint8)
+    {
+        case 1: octets[4] ipv4Address;
+        case 3: string domainName;
+        case 4: octets[16] ipv6Address;
```

Comment:
Why the gap in numbering? I would recommend 
```
        case 1: octets[4] ipv4Address;
        case 2: octets[16] ipv6Address;
        case 3: string domainName;
```
so it's like tcp only with just case 3 added.

Reply:
It is done like this to match the `ATYP` field specified in https://tools.ietf.org/html/rfc1928#section-4.

Reply:
I see, but I'm not sure if there is any value in that. @jfallows WDYT?